probe-rs/src/session.rs

```diff
@@ -298,8 +298,16 @@ impl Session {
             configured_trace_sink: None,
         };
 
+        for (core, _) in session.list_cores() {
```

Comment:
When working on the debugger, I found that `halt/continue` sometimes ran into a problem when:
1) `halt` ... but it was already halted because of a breakpoint.
2) `continue` ... this continues after the breakpoint, and to the user it looks like it didn't halt on the breakpoint.

Is this a concern at this stage?

Reply:
This happens very early in the attach process, and will also happen when flashing (running the flash detection sequence) but after that, it shouldn't happen. Therefore it shouldn't affect a debug session, once initiated?

Reply:
The ARM code actually halts the cores, and does not run them again. It would be nice to do the same here, to keep it consistent.

Reply:
Nevermind, that is only for attach under reset...